lib/insights/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -4,22 +4,32 @@ module Common
       module ApplicationControllerMixins
         module ExceptionHandling
           DEFAULT_ERROR_CODE = 400
+          PUNDIT_UNAUTHORIZED_ERROR = "Pundit::NotAuthorizedError".freeze
 
           def self.included(other)
             other.rescue_from(StandardError, RuntimeError) do |exception|
-              logger.error("#{exception.class.name}: #{exception.message}\n#{exception.backtrace.join("\n")}")
-              errors = Insights::API::Common::ErrorDocument.new.tap do |error_document|
-                exception_list_from(exception).each do |exc|
-                  if api_client_exception?(exc)
-                    api_client_errors(exc, error_document)
-                  else
-                    error_document.add(error_code_from_class(exc).to_s, "#{exc.class}: #{exc.message}")
-                  end
-                end
+              rescue_from_handler(exception) do |error_document, exc|
+                error_document.add(error_code_from_class(exc).to_s, "#{exc.class}: #{exc.message}")
               end
+            end
+          end
 
-              render :json => errors.to_h, :status => error_code_from_class(exception)
+          def rescue_from_handler(exception)
+            logger.error("#{exception.class.name}: #{exception.message}\n#{exception.backtrace.join("\n")}")
+            errors = Insights::API::Common::ErrorDocument.new.tap do |error_document|
+              exception_list_from(exception).each do |exc|
+                if api_client_exception?(exc)
+                  api_client_errors(exc, error_document)
+                elsif pundit_exception?(exc)
```

Comment:
It feels odd to have such a specific exception class in this generic class. More so because before this PR Pundit was handled in only the ApprovalApi / CatalogAPI apps.

Reply:
I wasn't really being handled in those apps, it was just falling through to the last `else` here, and the message that that case spits out was not very user friendly.